src/main/java/net/openhft/chronicle/queue/impl/single/SingleChronicleQueue.java

```diff
@@ -650,78 +619,80 @@ public long exactExcerptsInCycle(int cycle) {
     public long countExcerpts(long fromIndex, long toIndex) {
         throwExceptionIfClosed();
 
-        if (fromIndex > toIndex) {
-            long temp = fromIndex;
-            fromIndex = toIndex;
-            toIndex = temp;
-        }
-
-        // if the are the same
-        if (fromIndex == toIndex)
-            return 0;
+        try (ExcerptTailer tailer = createTailer()) {
```

Comment:
This is less garbage-y than it was, but it's garbage-y. I wonder if we should provide an overload that takes a Tailer as a zero-garbage option, and have this one call it with a tailer it creates and closes.

Reply:
This feels like a reasonable use case for a thread-local which is lazy instantiated then reset and used when needed.